src/main/java/com/labsynch/labseer/chemclasses/rdkit/CmpdRegMoleculeRDKitImpl.java

```diff
@@ -1,87 +1,69 @@
 package com.labsynch.labseer.chemclasses.rdkit;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
 
 import com.labsynch.labseer.chemclasses.CmpdRegMolecule;
+import com.labsynch.labseer.domain.RDKitStructure;
 import com.labsynch.labseer.exceptions.CmpdRegMolFormatException;
-import com.labsynch.labseer.utils.PropertiesUtilService;
 import com.labsynch.labseer.utils.SimpleUtil;
 
-import org.RDKit.KeyErrorException;
 import org.RDKit.RDKFuncs;
-import org.RDKit.ROMol;
 import org.RDKit.RWMol;
-import org.RDKit.Str_Vect;
 import org.codehaus.jackson.JsonNode;
 import org.codehaus.jackson.map.ObjectMapper;
 import org.codehaus.jackson.node.ObjectNode;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Configurable;
+import org.springframework.stereotype.Component;
 
-@Configurable
 public class CmpdRegMoleculeRDKitImpl implements CmpdRegMolecule {
 
 	Logger logger = LoggerFactory.getLogger(CmpdRegMoleculeRDKitImpl.class);
 
-	@Autowired
-	private PropertiesUtilService propertiesUtilService;
-
-	RWMol molecule;
-
 	static {
 		System.loadLibrary("GraphMolWrap");
 	}
+	
+	RDKitStructure molecule;
 
-	public CmpdRegMoleculeRDKitImpl(String molStructure) throws CmpdRegMolFormatException {
-		this.molecule = RWMol.MolFromMolBlock(molStructure);
-	}
+	private ExternalStructureService bbChemStructureService;
 
-	public CmpdRegMoleculeRDKitImpl(RWMol molecule) throws CmpdRegMolFormatException {
+	public CmpdRegMoleculeRDKitImpl(RDKitStructure molecule) {
 		this.molecule = molecule;
 	}
 
-	public CmpdRegMoleculeRDKitImpl(ROMol readOnlyMol) throws IOException, CmpdRegMolFormatException{
+	public CmpdRegMoleculeRDKitImpl(String mol, ExternalStructureService bbChemStructureService) {
+		this.bbChemStructureService = bbChemStructureService;
 		try {
-			this.molecule = new RWMol(readOnlyMol);
-		} catch(NullPointerException e) {
-			logger.error("Caught error trying to read mol");
-			throw new CmpdRegMolFormatException(e);
+			this.molecule = bbChemStructureService.getRDKitStructuresFromSDFService(mol).get(0);
```

Comment:
Does this service accept MOL input as well as SDF?
If it needs to be SDF, then should rename the `String mol` input in this function to `String sdf` or `String single_sdf` or something